cmd/clusterctl/cmd/move.go

```diff
@@ -64,13 +67,26 @@ func init() {
 		"The namespace where the workload cluster is hosted. If unspecified, the current context's namespace is used.")
 	moveCmd.Flags().BoolVar(&mo.dryRun, "dry-run", false,
 		"Enable dry run, don't really perform the move actions")
+	moveCmd.Flags().BoolVar(&mo.backupToFolder, "to-folder", false,
+		"Write Cluster API objects and all dependencies from a management cluster to folder.")
+	moveCmd.Flags().BoolVar(&mo.restoreFromFolder, "from-folder", false,
+		"Restore Cluster API objects from folder.")
+	moveCmd.Flags().StringVar(&mo.directory, "directory", "",
+		"The directory to save Cluster API objects to as yaml files")
+
+	moveCmd.MarkFlagsMutuallyExclusive("from-folder", "kubeconfig")
+	moveCmd.MarkFlagsMutuallyExclusive("to-folder", "to-kubeconfig")
+	moveCmd.MarkFlagsMutuallyExclusive("from-folder", "to-folder")
 
 	RootCmd.AddCommand(moveCmd)
 }
 
 func runMove() error {
 	// if no to kubeconfig provided and it's not a dry run, return error
-	if mo.toKubeconfig == "" && !mo.dryRun {
+	if !mo.backupToFolder &&
```

Comment:
Can you please also add a check to verify that if `--from-folder` or `--to-folder` are set to true then `--directory` cannot be emtpy?

Reply:
No problem. Implemented with cobra flags

```go
	moveCmd.MarkFlagsRequiredTogether("to-folder", "directory")
	moveCmd.MarkFlagsRequiredTogether("from-folder", "directory")
```

Reply:
Above didn't work so I added a prerun: 

```go
PreRun: func(cmd *cobra.Command, args []string) {
		toFolder, _ := cmd.Flags().GetBool("to-folder")
		fromFolder, _ := cmd.Flags().GetBool("from-folder")
		if toFolder || fromFolder {
			cmd.MarkFlagRequired("directory")
		}
	},
```

Now works as expected:
```shell
$ ./bin/clusterctl move --from-folder
Error: required flag(s) "directory" not set
$ ./bin/clusterctl move --to-folder
Error: required flag(s) "directory" not set
$ ./bin/clusterctl move --to-folder --directory /home/oscar/temp/
Performing backup...
```